### PR TITLE
Initial Scandit Capacitor Support

### PR DESCRIPTION
**Pull request by kdelorey**
### Summary
This PR adds initial support for a custom Scandit native capacitor plugin. This is not intended to be the final product as there is an initialization bug preventing me from completing it. Also it is intended that I merge the Image Scanners with the traditional Scanner plugins.

Right now this is just just to get it in the hands of QC. Currently it will only enable itself it the app is running within capacitor with the "ScanditNative" plugin available. Otherwise it'll fallback to its traditional behavior.

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/scandit-capacitor/scandit-capacitor.service.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    initialize(): Observable<string> {
+        return of(CapacitorPlugins.ScanditNative.initialize({
+            // todo: this is just a temporary trial key... needs to be changed when the initialization
+            // is working. This is relaltively safe for now.
+            apiKey: 'Afe/bGYVRrCYPHgW8TFeSSwGXJXGAYbWNmSmhplfA3pfehxgZFR1Krh6oj4EKQmn2F8D0yBxTgM6Hal4/G4kdYBeXJknZfBYWwTWANVSxJKbdH3Uf2KSxtRdEf4YQaWfVEuld6lQlrM+LpumXznPq8ItcngxL9D+FFAMeuD7YaKyN10S3cJ7ziZmwln+y9Cbc8KctlQt3XFMI84y2YDFsG/VBKn0ys6cE9HYKc1DET6GDGmCabylblsWEzgDRiYguUnH0YWZ5cCmc6nhk5sdUc0FtE6CgiM2G+bMl+KWBPWxCfE8Jeg6D9Cj+9nSTtuqHgVIHseN/hFiGkQvsAZEzJJWdspNJV5k5k9g+6uoBC2P2B+y8RU0wM9wYHeyQkNuVjlmpktPhz3jMR6IqRfyXZuxiSY7S4hmpAcIIO7gJLlSSd6YP8qd3C6s5Sv6hXJ0VihRH+oSL8MOL/b/b9WDvwqb2SsURdMx7Q1HYWH1Ng1BuEb/sbtO1+Bzk7hLXbKc4Txe1B49NWf6HtQMyy70CPqvrmx+NLxQOacehqVeselC3XvZcKt3GoO8CuaXU0ZnG8Xg8ibTZELy4yoaFPeXYkT7K1e/8HELhpn0clXm6XfQMAlsn2XNUfoJs+x9q0FyBeFb3NRxAsKtEpd+anShSSTtywHauZtuwDlM9lUbCj/l+oaZRPPsYImrhPMHI+NVHAP5B+i4LLw2yXqCdxBX+83ycsIQ5Rktm2DtXNpX0cJ1M0V4YdSw61bbQR+xnYDiVHVruw6Dh+mNOkWLpLJwz/AOLL46598f/AIsAEl+HAoiAglYpH7YLco0bAe1JZVSEDji3Naa7wM2'
```
Comment (dankaste) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/scandit-capacitor/scandit-capacitor.service.ts:29]: I assume you'll make this configuration driven in the final pass?

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { IPlatformPlugin } from '../platform-plugin.interface';
+import { IScanData } from '../scanners/scan.interface'
+
+export const IMAGE_SCANNERS = new InjectionToken<ImageScanner[]>('ImageScanners');
+
+export interface ScannerViewRef {
+    viewChanges(): Observable<{ left: number; top: number; width: number, height: number }>;
+}
+
+export interface ImageScanner extends IPlatformPlugin {
```
Comment (dankaste) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts:13]: I'm not sure image scanner would always be a platform plugin? There are some camera scanners that are full js I would assume we would want to make those work with ImageScanner as well

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/screen-parts/scan-or-search/scan-or-search.component.ts [hunk 3]
```diff
@@ -83,6 +89,18 @@ export class ScanOrSearchComponent extends ScreenPartComponent<ScanOrSearchInter
         super.ngOnDestroy();
     }
 
+    scan(data: IScanData) {
+        this.doAction(this.screenData.scanAction, data);
+    }
+
+    onScannerButtonClicked() {
+        if (this.imageScanners.isSupported) {
```
Comment (dankaste) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/screen-parts/scan-or-search/scan-or-search.component.ts:97]: are you still thinking you'll combine imageScanners and Scanners?

**Review by dankaste — CHANGES_REQUESTED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/scandit-capacitor/scandit-capacitor.service.ts (hunk 1 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/scandit-capacitor/scandit-capacitor.service.ts:29]: Right. This is all just temporary in order to get it in the hands of QC. The retrieval of configuration is actually what induced my initialization bug. It delayed it just enough to start having it happen.

This key is relatively safe as its bound to the native app and is also just a trial key that'll expire soon.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/screen-parts/scan-or-search/scan-or-search.component.ts (hunk 3 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/screen-parts/scan-or-search/scan-or-search.component.ts:97]: I am, I figured I'd keep the separated just for the time being. But I think I have a path figured out to combine them.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts (hunk 2 above)
Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts:13]: Yeah, I didn't really want this to be the case originally. I was running into a limitation. I'm going to take a closer look.

**Review by dankaste — APPROVED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts (hunk 2 above)
Comment (dankaste) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts:13]: I think you can just implement both interfaces in your specific plugin

Comment (kdelorey) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/platform-plugins/image-scanners/image-scanner.ts]: I removed the dependency.

**Pull request merged**